### PR TITLE
Improve `mackerel-agent configtest`: detect unexpected key

### PR DESCRIPTION
**Pull request by wafuwafu13**
part of: https://github.com/mackerelio/mackerel-agent/issues/810

#### example of WARNING

```
$ cat /usr/local/etc/mackerel-agent-sample.conf
apikey = "foo"
podfile = "/path/to/pidfile"

[foobar]
command = "test command"
env = { FOO = "BAR" }

[plugin.foo.bar]
command = "test command"
env = { FOO = "BAR" }

[plugin.metric.first]
command = "test command"

[plugin.check.first]
command = "test command"

[plugin.check.second]
command = "test command"

[plugins.check.first]
command = "test command"

[plugin.metrics.correct]
command = "test command"

[plugin.checks.correct]
command = "test command"
```

```zsh
~/desktop/mackerel-agent
$ ./build/mackerel-agent configtest -conf=/usr/local/etc/mackerel-agent-sample.conf
[WARNING] podfile is unexpected key  
[WARNING] foobar is unexpected key  
[WARNING] foobar.command is unexpected key  
[WARNING] foobar.env.FOO is unexpected key  
[WARNING] foobar.env is unexpected key  
[WARNING] plugins.check.first is unexpected key  
[WARNING] plugins.check.first.command is unexpected key  
[WARNING] plugin.metric.first is unexpected key  
[WARNING] plugin.check.first is unexpected key  
[WARNING] plugin.check.second is unexpected key  
[WARNING] plugin.foo.bar is unexpected key  
```

スクリーンショット 2022-09-12 16 47 52

#### example of CRITICAL

```
$ cat /usr/local/etc/mackerel-agent-sample-2.conf
uncorrectapikey = "foo"
```

```
~/desktop/mackerel-agent
$ ./build/mackerel-agent configtest -conf=/usr/local/etc/mackerel-agent-sample-2.conf
[CRITICAL] failed to test config: apikey must be specified in the config file (or by the DEPRECATED command-line flag) 
```

スクリーンショット 2022-09-12 16 45 55

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -93,24 +97,38 @@ func doVersion(_ *flag.FlagSet, _ []string) error {
...
-	configtest
+		configtest
 
 do configtest
 */
 func doConfigtest(fs *flag.FlagSet, argv []string) error {
 	conf, err := resolveConfig(fs, argv)
 	if err != nil {
-		return fmt.Errorf("failed to test config: %s", err)
+		return fmt.Errorf("\x1b[31m[CRITICAL] failed to test config: %s \x1b[0m", err)
```
Comment (lufia) [inline commands.go:110]: mackerel-agent runs on Windows, so we had better either to use https://github.com/mattn/go-colorable or https://github.com/fatih/color, or to disable ANSI escape codes except on `go:build unix`.

Comment (wafuwafu13) [reply commands.go]: fixed https://github.com/mackerelio/mackerel-agent/pull/813/commits/56a06425e7c9bb2803ed01b2b00119b2b5d95d3e

File: config/validate.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+// ValidateConfigFile detect unexpected key in configfile
+func ValidateConfigFile(file string) ([]string, error) {
+	config := &Config{}
+	md, err := toml.DecodeFile(file, config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to test config: %s", err)
+	}
+
+	var unexpectedKey []string = make([]string, 0)
```
[the remark this answers is not in the document]
Comment (wafuwafu13) [reply config/validate.go]: fixed https://github.com/mackerelio/mackerel-agent/pull/813/commits/0554f6b160bca9ea2fc1c1fde1e371d97c801873

**Review by lufia — APPROVED** [review]
Thanks!

**Pull request merged**